src/entry/mod.rs

```diff
@@ -0,0 +1,15 @@
+use hopper;
+use metric::Event;
+
+pub trait Entry {
```

Comment:
It occurs to me that the `Entry` concept has a conceptual problem, I think. We don't really _need_ the full configuration as received from the users, only the bits that are important to the operation of the sink. That makes the ownership model of the config straightforward: we move the pieces of use into the sink and drop the rest. `Entry` retains ownership of the config – necessitating cloning – but I'm not sure _why_ this is needed. `Entry` finds use in cernan.rs for building up the forwards but the implementation is less straightforward than the approach in #213. Testing #213 is hard and maybe an explicit `Entry` helps with that? 

Also, keeping `Entry` as such in memory is problematic for credentials, especially those systems where it's possible to compute a session key from long-lived credentials. Right now we've got a design that'll allow us to dump those long-lived credentials through deallocation but `Entry` – I think – would lock us into keeping them in-memory for the lifetime of the cernan instance.